src/dune_rules/jsoo/jsoo_rules.ml

```diff
@@ -411,17 +445,21 @@ let build_exe cc ~loc ~in_context ~src ~(obj_dir : Path.Build.t Obj_dir.t)
     | Some p -> Promote p
   in
   let open Memo.O in
-  let* cmode = Super_context.js_of_ocaml_compilation_mode sctx ~dir in
-  match (cmode : Js_of_ocaml.Compilation_mode.t) with
-  | Separate_compilation ->
+  let* cmode = Super_context.js_of_ocaml_compilation_mode sctx ~dir
+  and* ctarget = Super_context.js_of_ocaml_target sctx ~dir in
+  match (cmode, ctarget) with
+  | Separate_compilation, JS ->
     standalone_runtime_rule cc ~javascript_files ~target:standalone_runtime
       ~flags
     >>= Super_context.add_rule ~loc sctx ~dir
     >>> link_rule cc ~runtime:standalone_runtime ~target ~obj_dir
           top_sorted_modules ~flags ~linkall ~link_time_code_gen
     >>= Super_context.add_rule sctx ~loc ~dir ~mode
-  | Whole_program ->
-    exe_rule cc ~javascript_files ~src ~target ~flags
+  | Whole_program, JS ->
+    exe_rule ~ctarget cc ~javascript_files ~wasm_files:[] ~src ~target ~flags
+    >>= Super_context.add_rule sctx ~loc ~dir ~mode
+  | _, Wasm ->
+    exe_rule ~ctarget cc ~javascript_files ~wasm_files ~src ~target ~flags
```

Comment:
Shouldn't it be an error if `Separate_compilation, Wasm` ? instead of silently ignoring cmode.

Reply:
The idea is that you can switch to producing Wasm code just by setting the target, without having to specify a compilation mode as well.
Do you think it would be better to change `Super_context.js_of_ocaml_compilation_mode` to not default to `Separate_compilation` when the target is `wasm`?

Reply:
Have you considered making wasm part of `compilation_mode` ?
I'm also happy to keep this as is and revisit later if needed.

Reply:
Leave it as is then.

Reply:
Maybe add a comment in the code saying that separate compilation is coming later.